Mathlib/Algebra/GroupWithZero/Hom.lean

```diff
@@ -234,6 +239,13 @@ lemma one_apply_of_ne_zero {M₀ N₀ : Type*} [MulZeroOneClass M₀] [MulZeroOn
     (1 : M₀ →*₀ N₀) x = 1 :=
   if_neg hx
 
+@[simp]
+lemma one_apply_eq_zero_iff {M₀ N₀ : Type*} [MulZeroOneClass M₀] [MulZeroOneClass N₀]
+    [DecidablePred fun x : M₀ ↦ x = 0] [Nontrivial M₀] [NoZeroDivisors M₀] [NeZero (1 : N₀)]
+    {x : M₀} :
+    (1 : M₀ →*₀ N₀) x = 0 ↔ x = 0 := by
```

Comment:
[NeZero (1 : N₀)] can be replaced by Nontrivial N₀.
(Same for `one_apply_eq_zero/lt_one_iff` below)